source/Halibut/HalibutRuntime.cs

```diff
@@ -216,6 +221,20 @@ public void Dispose()
             }
         }
 
+        protected bool NotifyUnAuthorizedClientConnect(string clientName, string thumbPrint)
```

Comment:
This returning a `bool` doesn't sit right with me. The name of the method doesn't really help me understand what a return value of `true` or `false` actually means.
Can we use another enum? `BlockConnection`/`AllowConnection`?

Reply:
I changed it to use the HandleUnauthorizedClientMode enum instead, and changed the name to OnUnauthorizedClientConnect as it's really firing an event if assigned.